cmd/process-agent/collector.go

```diff
@@ -23,9 +23,25 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/clustername"
 )
 
+type checkResult struct {
+	name        string
+	payloads    []checkPayload
+	sizeInBytes int64
+}
+
+func (cr *checkResult) Weight() int64 {
+	return cr.sizeInBytes
+}
+
+func (cr *checkResult) Type() string {
+	return cr.name
+}
```

Comment:
Question: is this type really needed?

Reply:
In what sense?  It needs to implement the `WeightedItem` interface.

Reply:
Yea I was asking more of regarding whether we need Type for the interface but I forgot the fact that process & containers are in the same queue.